src/appleseed/renderer/kernel/rendering/final/adaptivetilerenderer.cpp

```diff
@@ -270,23 +265,22 @@ namespace
             assert(framebuffer);
 
             // Create the buffer into which we will accumulate every second samples.
-            ShadingResultFrameBuffer* second_framebuffer = nullptr;
-
             // If rendering multiple passes, the permanent buffer factory will return
             // the same buffer so we must create a new one.
-            second_framebuffer = new ShadingResultFrameBuffer(
-                tile.get_width(),
-                tile.get_height(),
-                frame.aov_images().size(),
-                tile_bbox,
-                frame.get_filter());
+            unique_ptr<ShadingResultFrameBuffer> second_framebuffer(
+                new ShadingResultFrameBuffer(
+                    tile.get_width(),
+                    tile.get_height(),
+                    frame.aov_images().size(),
+                    tile_bbox,
+                    frame.get_filter()));
 
             if (m_params.m_passes > 1)
                 second_framebuffer->copy_from(*framebuffer);
             else
                 second_framebuffer->clear();
 
-            assert(second_framebuffer);
+            assert(second_framebuffer.get());
```

Comment:
That assertion looks redundant: I don't see how `second_framebuffer` can be null, and you use it before the assertion anyway. I would suggest to remove it.